lib/record_store/provider/ns1/patch_api_header.rb

```diff
@@ -18,6 +18,10 @@ def process_response(response)
         rate_limit.wait(sleep_time)
       end
 
+      if Net::HTTPResponse::CODE_TO_OBJ[response.code] == Net::HTTPServiceUnavailable
+        return NS1::Response::Error.new({}, response.code.to_i)
+      end
```

Comment:
I would suggest replacing this by a `begin..end` around the `JSON.parse` to raise the `UnparsableError` I talked about above. That way the response code can stay the same. Maybe the `UnparsableError` could be part of the patch here (`class NS1::Response::UnparsableError < NS1::Response::Error; end`) ?

Reply:
That's really clean, thanks for the suggestion!